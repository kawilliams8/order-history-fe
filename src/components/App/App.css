.App {
  text-align: center;
}

header {
  align-items: center;
  background: url('../../images/checkout.jpg') no-repeat center / 100%;
  color: #FFF;
  display: flex;
  flex-direction: row-reverse;
  justify-content: space-around;
  height: 300px;
}

h1 {
  font-size: 72px;
}

.purchase-form {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  height: 100%;
  width: 50%;
}

.purchase-container {
  height: 700px;
  background:#CECECE;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}