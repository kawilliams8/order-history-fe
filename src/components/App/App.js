import React, { Component } from 'react';
import '../App/App.css';
import Orders from '../../components/Orders/Orders';
import OrderForm from '../OrderForm/OrderForm';

class App extends Component {
  constructor() {
    super()
    this.state = {
      orders: [],
      error: ""
    };
  }

  componentDidMount() {
    fetch('http://localhost:3001/api/v1/purchases')
    .then(response => response.json())
    .then(data => this.setState({orders: data}))
    .catch(error => this.setState({error: error.message}))
  }

  addOrder = (newOrder) => {
    const options = {
      method: "POST",
      body: JSON.stringify(newOrder),
      headers: {
        'Content-Type': 'application/json'
      }
    }

    fetch('http://localhost:3001/api/v1/purchases', options)
    .then(response => response.json())
    .then(data => this.setState({orders: data}))
    .catch(error => this.setState({ error: error.message }))
  }

  removeOrder = (id) => {
    const options = {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json'
      }
    }
    
    fetch(`http://localhost:3001/api/v1/purchases/${id}`, options)
    .then(() => fetch('http://localhost:3001/api/v1/purchases'))
    .then(response => response.json())
    .then(data => this.setState({orders: data}))
    .catch(error => this.setState({ error: error.message }))
  }

  render() {
    return (
      <div className="App">
        <header>
          <h1 className='app-title'>My Order History</h1>
          <div className='purchase-form'>
            <OrderForm addOrder={this.addOrder}/>
          </div>
        </header>
        <div className='purchase-container'>
          <Orders data={this.state.orders} removeOrder={this.removeOrder}/>
        </div>
      </div>
    );
  }
}

export default App;
