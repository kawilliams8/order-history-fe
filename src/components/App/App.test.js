import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
  });

  it('should update state when addOrder is called', async () => {
    const wrapper = await shallow(<App />);
    const mockOrder = {
      id: 1,
      img: "image.png",
      name: "Mock Order",
      description: "This is a mock description",
      price: 3
      }

    const expected = [mockOrder];
    expect(wrapper.state('orders')).toEqual([]);
    await(wrapper.instance().addOrder(mockOrder))
    expect(wrapper.state('orders')).toEqual(expected)
    })
  
})