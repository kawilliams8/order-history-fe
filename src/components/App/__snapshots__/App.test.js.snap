// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <header>
    <h1
      className="app-title"
    >
      My Order History
    </h1>
    <div
      className="purchase-form"
    >
      <OrderForm
        addOrder={[Function]}
      />
    </div>
  </header>
  <div
    className="purchase-container"
  >
    <Orders
      data={Array []}
      removeOrder={[Function]}
    />
  </div>
</div>
`;
